src/coreclr/jit/gentree.cpp

```diff
@@ -21308,6 +21308,158 @@ GenTree* Compiler::gtNewSimdCeilNode(var_types type, GenTree* op1, CorInfoType s
     return gtNewSimdHWIntrinsicNode(type, op1, intrinsic, simdBaseJitType, simdSize);
 }
 
+#if defined(TARGET_XARCH)
+GenTreeVecCon* Compiler::gtCvtCtrlTbl(var_types type, var_types sourceType, var_types targetType, unsigned simdSize)
+{
+    assert(IsBaselineSimdIsaSupportedDebugOnly());
+    assert(IsBaselineVector512IsaSupportedDebugOnly());
+    assert(varTypeIsFloating(sourceType));
+    assert(varTypeIsIntegral(targetType));
+    assert(varTypeIsSIMD(type));
+    assert(getSIMDTypeForSize(simdSize) == type);
+
+    GenTreeVecCon* tbl = gtNewVconNode(type);
+
+    switch (sourceType)
+    {
+        case TYP_DOUBLE:
+            switch (targetType)
+            {
+                case TYP_UINT:
+                case TYP_ULONG:
+                    for (int i = 0; i < 8; i++)
+                    {
+                        tbl->gtSimdVal.i64[i] = 0x08080088;
+                    }
+                    break;
+
+                case TYP_INT:
+                case TYP_LONG:
+                    for (int i = 0; i < 8; i++)
+                    {
+                        tbl->gtSimdVal.i64[i] = 0x00000088;
+                    }
+                    break;
+
+                default:
+                    unreached();
+            }
+            break;
+
+        case TYP_FLOAT:
+            switch (targetType)
+            {
+                case TYP_UINT:
+                case TYP_ULONG:
+                    for (int i = 0; i < 16; i++)
+                    {
+                        tbl->gtSimdVal.i32[i] = 0x08080088;
+                    }
+                    break;
+
+                case TYP_INT:
+                case TYP_LONG:
+                    for (int i = 0; i < 16; i++)
+                    {
+                        tbl->gtSimdVal.i32[i] = 0x00000088;
+                    }
+                    break;
+
+                default:
+                    unreached();
+            }
+            break;
+
+        default:
+            unreached();
+    }
+    return tbl;
+}
+
+GenTree* Compiler::gtNewSimdCvtNode(var_types      type,
+                                    GenTree*       op1,
+                                    NamedIntrinsic hwIntrinsicID,
+                                    CorInfoType    simdTargetBaseJitType,
+                                    CorInfoType    simdSourceBaseJitType,
+                                    unsigned       simdSize)
+{
+    assert(varTypeIsSIMD(type));
+    assert(getSIMDTypeForSize(simdSize) == type);
+    assert(op1 != nullptr);
+    assert(op1->TypeIs(type));
+
+    var_types simdSourceBaseType = JitType2PreciseVarType(simdSourceBaseJitType);
+    var_types simdTargetBaseType = JitType2PreciseVarType(simdTargetBaseJitType);
+    assert(varTypeIsArithmetic(simdSourceBaseType));
+    assert(varTypeIsArithmetic(simdTargetBaseType));
```

Comment:
We shouldn't need these, they are a superset of the `varTypeIsFloating` and `varTypeIsIntegral` checks just below.

Reply:
True. I had them for my own sake during development. removing them in the latest code.